src/components/JoinExistingList.jsx

```diff
@@ -0,0 +1,53 @@
+import { db } from '../lib/firebase';
+import { collection, onSnapshot } from '@firebase/firestore';
+import { useState, useEffect } from 'react';
+import { setUser } from '../storage-utils/storage-utils';
+import { useNavigate } from 'react-router-dom';
+
+export default function JoinExistingList() {
+  const [userToken, setUserToken] = useState('');
+  const [docs, setDocs] = useState([]);
+  const navigate = useNavigate();
+
+  // get the collection when userToken changes
+  useEffect(() => {
+    if (userToken !== '') {
+      const unsubscribe = onSnapshot(collection(db, userToken), (snapshot) => {
+        let snapshotDocs = [];
+        snapshot.forEach((doc) => snapshotDocs.push(doc));
+        setDocs(snapshotDocs);
+      });
+      return () => {
+        unsubscribe();
+      };
+    }
+  }, [userToken]);
+
+  return (
+    <div>
+      <p>Join an existing list by entering a three word token below.</p>
+
+      <form
+        onSubmit={() => {
+          if (docs.length > 0) {
```

Comment:
Should we instead validate that the input is three words?

Reply:
Would something like this make more sense?
![Screen Shot 2022-04-22 at 5 51 54 PM](https://user-images.githubusercontent.com/67212475/164844125-7bd38da0-c6e2-4e03-8e97-c07f3fcefdf3.png)

Reply:
Refactoring a little I found we didn't need to check against docs at all, this seemingly works just the same as before but with way less code and now the conditional check on submit just checks for invalid token input. 

Does this better suit our ticket?

![Screen Shot 2022-04-22 at 7 34 36 PM](https://user-images.githubusercontent.com/67212475/164867729-19f935b4-aefc-46a9-8905-770389a436ef.png)

Reply:
We're having trouble finding a way to do that without the check against the docs we have saved in state. Firebase doesn't allow a grab of all collections (so we can loop thru for id's + tokens, etc) due to security risks. Do you have something in mind we could implement?

Reply:
We thought of that too but  think it leaves a bit too much room for error? what if someone enters a 3 word token but mispells a word? We came to the code we have because it checks if a collection associated with the entered token exists. If it doesn't exist, the call to the firestore collection returns an empty docs object.

Reply:
Ah, so there are two bits here (and it's okay if you don't feel it's necessary to implement both):

1. input validation (check input conforms to expected) - this doesn't require any API calls
2. valid token (check that token exists in firestore) - you're doing this, great!

The advantage of checking/doing both is that you can avoid unnecessary calls to an external API.

Reply:
Ah! That makes sense! Thank you for clarifying 😄